artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnector.java

```diff
@@ -584,15 +585,15 @@ public synchronized void start() {
             realTrustStorePassword = trustStorePassword;
          } else {
             realKeyStorePath = Stream.of(System.getProperty(ACTIVEMQ_KEYSTORE_PATH_PROP_NAME), System.getProperty(JAVAX_KEYSTORE_PATH_PROP_NAME), keyStorePath).map(v -> useDefaultSslContext ? keyStorePath : v).filter(Objects::nonNull).findFirst().orElse(null);
-            realKeyStorePassword = Stream.of(System.getProperty(ACTIVEMQ_KEYSTORE_PASSWORD_PROP_NAME), System.getProperty(JAVAX_KEYSTORE_PASSWORD_PROP_NAME), keyStorePassword).map(v -> useDefaultSslContext ? keyStorePassword : v).filter(Objects::nonNull).findFirst().orElse(null);
+            realKeyStorePassword = processSslPasswordProperty(Stream.of(System.getProperty(ACTIVEMQ_KEYSTORE_PASSWORD_PROP_NAME), System.getProperty(JAVAX_KEYSTORE_PASSWORD_PROP_NAME), keyStorePassword).map(v -> useDefaultSslContext ? keyStorePassword : v).filter(Objects::nonNull).findFirst().orElse(null));
```

Comment:
The system properties have a grater visibility than any connector properties and multiple connector could rely on the same system properties so it doesn't sound good to me that a connector property affects the use of a system property.

Reply:
The stream processing logic is identical to the previous code--I just wrapped the stream result in a method call to handle the possibility of an encrypted system property. However, the encrypted password logic should only be invoked when the value is not the keyStorePassword or trustStorePassword captured with ConfigurationHelper#getPasswordProperty(...) previously as that value has already been decrypted if necessary.

I've tweaked the logic to call processSslPasswordProperty(...) only when the existing stream processing result is one of the system properties, not a query string value.

Reply:
I'm not sure to get the purpose of this PR, why does a user need to set the password-codec in the connector url and the passwords in system properties?

Reply:
This PR enables ENC(...) handling for the 'org.apache.activemq.ssl.keyStorePassword', 'o.a.a.ssl.trustStorePassword', 'javax.net.ssl.keyStorePassword', and 'j.n.ssl.trustStorePassword' system property values. The connector URL 'keyStorePassword' and 'trustStorePassword' query string handling is unchanged as these values already are decrypted as needed through the ConfigurationHelper#getPasswordProperty(...) method in lines 401 and 409.

The system properties obviously aren't part of the 'configuration' Map so the ConfigurationHelper#getPasswordProperty(...) method can't be used directly. The processSslPasswordProperty(...) method duplicates the behavior of calling getPasswordProperty(...) using the same ActiveMQDefaultConfiguration.getPropMaskPassword() and .getPropPasswordCodec() used for the connector URL password handling for the 'keyStorePassword' and 'trustStorePassword'.

Hopefully, that makes more sense as to the intent. ARTEMIS-3794 is just a JIRA I saw similar to the AMQP broker connection user and password attribute encryption handling.

Reply:
If the purpose is to override keyStore/trustStore passwords obtained from topology with encrypted system properties maybe we should use system properties to set the mask mode and the password codec too, i.e. 'org.apache.activemq.ssl.passwordCodec', the server and the client could use a different mask mode or password codec.

Reply:
That's an interesting thought. I based these updates on the approach handling keyStorePassword and trustStorePassword also applying adequately for their corresponding system property values.

I'm all for providing essential flexibility, but I've also never had to provide an alternate password encoding codec to appease PEN testing or production environment security scans. I'm happy to address that if you see the need for that additional configuration and associated incremental complexity.

Perhaps we can see if anyone files an enhancement request for specifying an alternate mask mode and/or password codec with system properties?